File: bulk/csharp/ZipSlip01.cs
// CWE-22: Zip Slip
using System.IO;
using System.IO.Compression;
public class ZipSlip01 {
    public void Extract(string zipPath, string destDir) {
        using var archive = ZipFile.OpenRead(zipPath);
        foreach (var entry in archive.Entries) {
            var destPath = Path.Combine(destDir, entry.FullName); // no path check
            entry.ExtractToFile(destPath, true);
        }
    }
}

File: bulk/java/ZipSlip01.java
// CWE-22: Zip Slip
import java.io.*;
import java.util.zip.*;
public class ZipSlip01 {
    public void extract(ZipFile zip, File destDir) throws IOException {
        zip.entries().asIterator().forEachRemaining(entry -> {
            File out = new File(destDir, entry.getName()); // no path check
            try { new FileOutputStream(out); } catch (IOException e) {}
        });
    }
}
